frontend/src/app/modules/availability/availability-page/availability-page.component.ts

```diff
@@ -1,8 +1,57 @@
 import { Component } from '@angular/core';
+import { AvailabilitySlot } from '@core/models/availiability-slot';
+import { TeamWithSlots } from '@core/models/team-with-slot';
+import { User } from '@core/models/user';
+import { AvailabilitySlotService } from '@core/services/availability-slot.service';
+import { UserService } from '@core/services/user.service';
 
 @Component({
     selector: 'app-availability-page',
     templateUrl: './availability-page.component.html',
     styleUrls: ['./availability-page.component.sass'],
 })
-export class AvailabilityPageComponent {}
+export class AvailabilityPageComponent {
+    public teams: Array<TeamWithSlots>;
+
+    public currentUser: User;
+
+    public userSlots: Array<AvailabilitySlot>;
+
+    constructor(private availabilitySlotService: AvailabilitySlotService, private userService: UserService) {
+        this.getCurrentUser();
+    }
+
+    public getAllAvailabilitySlotsForTeam() {
+        this.availabilitySlotService
+            .getAvailabilitySlotsGroupByTeams(this.currentUser.id)
+            .subscribe(
+                (resp) => {
+                    this.teams = resp;
+                },
+            );
+    }
+
+    public getAllUsersAvailabilitySlots() {
+        this.availabilitySlotService
+            .getAllUsersAvailabilitySlots(this.currentUser.id)
+            .subscribe(
+                (resp) => {
+                    this.userSlots = resp;
+                },
+            );
+    }
+
+    public getCurrentUser() {
+        this.userService
+            .getCurrentUserById(4)
```

Comment:
It would be better to not hard code the number
At least we need to move it to a separate variable